src/Engine/Manager/RenderManager.cpp

```diff
@@ -283,6 +257,63 @@ void RenderManager::Render(World& world, const glm::mat4& viewMatrix, const glm:
 
 }
 
+void RenderManager::RenderEditorEntities(World& world, bool soundSources, bool particleEmitters, bool lightSources,
+    bool cameras, bool physics, const glm::vec3& position, const glm::mat4& viewMatrix, const glm::mat4& projectionMatrix,
+    Video::RenderSurface* renderSurface) {
+
+    const glm::mat4 viewProjectionMatrix = projectionMatrix * viewMatrix;
+    const glm::vec3 up(viewMatrix[0][1], viewMatrix[1][1], viewMatrix[2][1]);
+
+    renderSurface->GetShadingFrameBuffer()->BindWrite();
+    renderer->PrepareRenderingIcons(viewProjectionMatrix, position, up);
+
+    // Render sound sources.
+    if (soundSources) {
+        for (SoundSource* soundSource : Managers().soundManager->GetSoundSources())
+            renderer->RenderIcon(soundSource->entity->GetWorldPosition(), soundSourceTexture);
+    }
+
+    // Render particle emitters.
+    if (particleEmitters) {
+        for (ParticleEmitter* emitter : Managers().particleManager->GetParticleEmitters())
+            renderer->RenderIcon(emitter->entity->GetWorldPosition(), particleEmitterTexture);
+    }
+
+    // Render light sources.
+    if (lightSources) {
+        for (DirectionalLight* light : directionalLights.GetAll())
+            renderer->RenderIcon(light->entity->GetWorldPosition(), lightTexture);
+
+        for (PointLight* light : pointLights.GetAll())
+            renderer->RenderIcon(light->entity->GetWorldPosition(), lightTexture);
+
+        for (SpotLight* light : spotLights.GetAll())
+            renderer->RenderIcon(light->entity->GetWorldPosition(), lightTexture);
+    }
+
+    // Render cameras.
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Variable 'name' is assigned a value that is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)